geomstats/geometry/spd_matrices.py

```diff
@@ -7,50 +7,27 @@
 from geomstats.geometry.general_linear import GeneralLinear
 from geomstats.geometry.matrices import Matrices
 from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.geometry.symmetric_matrices import SymmetricMatrices
 
 EPSILON = 1e-6
 TOLERANCE = 1e-12
 
 
-class SPDMatrices(EmbeddedManifold):
+class SPDMatrices(SymmetricMatrices, EmbeddedManifold):
     """Class for the manifold of symmetric positive definite (SPD) matrices."""
 
     def __init__(self, n):
         super(SPDMatrices, self).__init__(
+            n=n,
             dimension=int(n * (n + 1) / 2),
             embedding_manifold=GeneralLinear(n=n))
-        self.n = n
```

Comment:
I don't know who was involved in this decision but I read @opeltre in #413:
> 1. So as we agreed SPD(n) should not inherit from GL(n) as it is not a subgroup of the latter.

I think @opeltre had in mind in his overview.md file ([read here](https://github.com/opeltre/geomstats/blob/doc-overview/OVERVIEW.md))to put exp and log in SPD. However I agree with the viewpoint that `expm` should be in SymmetricMatrices and `logm` in SPD because this embodies the diffeo between the two spaces.
About inheritance or embedding for inclusion, I think there is currently a mix of the two in geomstats, especially with `LieGroups` that don't necessarily inherit from `GeneralLinear` (while they should inherit the compose, inv, etc methods) but also with `LieAlgebra, Matrices, SymmetricMatrices`, etc..

Reply:
Originally, we used the logic "A is a special case of B" to determine if A should inherit from B, and not the inclusion. 

If we start using the inclusion of SPD in Sym for inheritance, maybe we should do it everywhere?

To inherit methods of GLn, we could also use the fact that MatrixLieGroups are subgroups of GLn and implement GLn's methods there.